CRM/Event/BAO/Participant.php

```diff
@@ -1893,6 +1893,17 @@ public static function getSelfServiceEligibility(int $participantId, string $url
    * @throws CRM_Core_Exception
    */
   public static function self_hook_civicrm_pre(\Civi\Core\Event\PreEvent $event) {
+    // Set the default role ID on create.
+    if ($event->entity === 'Participant' && $event->action === 'create' && empty($event->params['role_id'])) {
```

Comment:
The `$event->entity === 'Participant'` clause is redundant because this is a `self_` hook callback.